ci/setup_env.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash -e
+
+
+# edit the locale file if needed
+if [ -n "$LOCALE_OVERRIDE" ]; then
+    echo "Adding locale to the first line of pandas/__init__.py"
+    rm -f pandas/__init__.pyc
+    SEDC="3iimport locale\nlocale.setlocale(locale.LC_ALL, '$LOCALE_OVERRIDE')\n"
+    sed -i "$SEDC" pandas/__init__.py
+    echo "[head -4 pandas/__init__.py]"
+    head -4 pandas/__init__.py
+    echo
+    sudo locale-gen "$LOCALE_OVERRIDE"
+fi
+
+MINICONDA_DIR="$HOME/miniconda3"
+
+
+if [ -d "$MINICONDA_DIR" ]; then
+    echo
+    echo "rm -rf "$MINICONDA_DIR""
+    rm -rf "$MINICONDA_DIR"
+fi
+
+echo "Install Miniconda"
+UNAME_OS=$(uname)
+if [[ "$UNAME_OS" == 'Linux' ]]; then
+    if [[ "$BITS32" == "yes" ]]; then
+        CONDA_OS="Linux-x86"
+    else
+        CONDA_OS="Linux-x86_64"
+    fi
+elif [[ "$UNAME_OS" == 'Darwin' ]]; then
+    CONDA_OS="MacOSX-x86_64"
+else
+  echo "OS $UNAME_OS not supported"
+  exit 1
+fi
+
+wget -q "https://repo.continuum.io/miniconda/Miniconda3-latest-$CONDA_OS.sh" -O miniconda.sh
+chmod +x miniconda.sh
+./miniconda.sh -b
+
+export PATH=$MINICONDA_DIR/bin:$PATH
+
+echo
+echo "which conda"
+which conda
+
+echo
+echo "update conda"
+conda config --set ssl_verify false
+conda config --set quiet true --set always_yes true --set changeps1 false
+conda update -n base conda
+
+echo "conda info -a"
+conda info -a
+
+echo
+echo "set the compiler cache to work"
+if [ -z "$NOCACHE" ] && [ "${TRAVIS_OS_NAME}" == "linux" ]; then
+    echo "Using ccache"
+    export PATH=/usr/lib/ccache:/usr/lib64/ccache:$PATH
+    GCC=$(which gcc)
+    echo "gcc: $GCC"
+    CCACHE=$(which ccache)
+    echo "ccache: $CCACHE"
+    export CC='ccache gcc'
+elif [ -z "$NOCACHE" ] && [ "${TRAVIS_OS_NAME}" == "osx" ]; then
+    echo "Install ccache"
+    brew install ccache > /dev/null 2>&1
+    echo "Using ccache"
+    export PATH=/usr/local/opt/ccache/libexec:$PATH
+    gcc=$(which gcc)
+    echo "gcc: $gcc"
+    CCACHE=$(which ccache)
+    echo "ccache: $CCACHE"
+else
```

Comment:
@datapythonista I think nowadays this caching might be counter-productive, though needs some testing, if you want to make an issue to remove it (but don't touch here)